### PR TITLE
Added Technologies

### PR DESCRIPTION
**Pull request by abdullah-dev100**
- Added rspec for technologies
- Changed module name of technologies
- Change module to class
- Added byebug gem for debugging
- Added methods for technology management

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-gem "rake", "~> 13.0"
+gem 'rake', '~> 13.0'
 
-gem "rspec", "~> 3.0"
+gem 'rspec', '~> 3.0'
 
-gem "rubocop", "~> 1.21"
+gem 'rubocop', '~> 1.21'
+
+gem 'byebug', platforms: %i[mri mingw x64_mingw]
```
Comment (ziaulrehman40) [inline Gemfile]: should they only be in development? Otheriwse they will install this in prod too

File: README.md [hunk 2]
```diff
@@ -18,6 +18,9 @@ If bundler is not being used to manage dependencies, install the gem by executin
 
 TODO: Write usage instructions here
 
+- This gem only lists the technologies and does not create or update the existing technologies
+- To list all technologies use -> MapleTech::Technologies.all
```
Comment (ziaulrehman40) [inline README.md:22]: need to review it visually

File: lib/maple_tech/technologies/version.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module MapleTech
+  class Technologies
+    VERSION = '0.1'
+  end
```
Comment (ziaulrehman40) [inline lib/maple_tech/technologies/version.rb:6]: why is this needed?

File: spec/maple_tech/technologies_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+RSpec.describe(MapleTech::Technologies) do
+  context 'version' do
+    it 'has a version number' do
+      expect(MapleTech::Technologies::VERSION).not_to(be(nil))
+    end
+
+    it 'must has version number of 0.1.0' do
+      expect(MapleTech::Technologies::VERSION).to(eql('0.1'))
+    end
```
Comment (ziaulrehman40) [inline spec/maple_tech/technologies_spec.rb]: this is needless spec

File: spec/maple_tech/technologies_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  context 'technologies' do
+    it 'should not have nil value' do
+      nil_value = MapleTech::Technologies.all.any? { |value| value.nil? }
+      expect(nil_value).to(eql(false))
+    end
+
+    it 'should not have empty value' do
+      empty_value = MapleTech::Technologies.all.any? { |value| value.eql?('') }
+      expect(empty_value).to(eql(false))
+    end
```
Comment (ziaulrehman40) [inline spec/maple_tech/technologies_spec.rb:19]: should not have any duplicates

File: spec/maple_tech/technologies_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  end
```
Comment (ziaulrehman40) [inline spec/maple_tech/technologies_spec.rb:34]: should not have case in-sensitive duplicates either

File: technologies-list-maple.gemspec [hunk 7]
```diff
@@ -1,39 +1,41 @@
...
+require_relative 'lib/maple_tech/technologies/version'
 
 Gem::Specification.new do |spec|
-  spec.name = "technologies-list-maple"
-  spec.version = Technologies::List::Maple::VERSION
-  spec.authors = ["Muhammad Abdullah"]
-  spec.email = ["dev@Devs-MacBook-Pro.local"]
+  spec.name = 'technologies-list-maple'
+  spec.version = MapleTech::Technologies::VERSION
+  spec.authors = ['Muhammad Abdullah']
```
Comment (ziaulrehman40) [inline technologies-list-maple.gemspec]: MapleHR

File: technologies-list-maple.gemspec [hunk 8]
```diff
@@ -1,39 +1,41 @@
...
+  spec.email = ['muhammad.abdullah@devsinc.com']
```
Comment (ziaulrehman40) [inline technologies-list-maple.gemspec]: engineering@maplehr.io

File: lib/maple_tech/technologies/version.rb (hunk 3 above)
Comment (abdullah-dev100) [reply lib/maple_tech/technologies/version.rb:6]: to maintain versions, by default gems add a versions.

File: Gemfile [hunk 9]
```diff
@@ -1,12 +1,16 @@
...
 
-gem "rake", "~> 13.0"
+gem 'rake', '~> 13.0'
 
-gem "rspec", "~> 3.0"
+gem 'rspec', '~> 3.0'
 
-gem "rubocop", "~> 1.21"
+gem 'rubocop', '~> 1.21'
+
```
Comment (ziaulrehman40) [inline Gemfile]: not moved

File: spec/maple_tech/technologies_spec.rb [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    it 'should not have empty value' do
+      empty_value = MapleTech::Technologies.all.any? { |value| value.eql?('') }
+      expect(empty_value).to(eql(false))
+    end
+
+    it 'should not have duplicate values' do
+      uniq_count = MapleTech::Technologies.all.uniq(&:downcase).length
+      total_technologies_count = MapleTech::Technologies.all.length
+      expect(uniq_count).to(eql(total_technologies_count))
+    end
```
Comment (ziaulrehman40) [inline spec/maple_tech/technologies_spec.rb:33]: failing spec should excatly tell duplicate values, instead of relying on user to identify duplicates

File: spec/maple_tech/technologies_spec.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  end
```
Comment (ziaulrehman40) [inline spec/maple_tech/technologies_spec.rb:34]: also add spec that no technology should have spaces at start or end

**Review by Musharraf-2 — APPROVED** [review]

**Review by Hassan-lateef — APPROVED** [review]
